src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,88 @@
+package duke;
+
+import java.util.ArrayList;
+
+
+/**
+ * TaskList class is an extension of the java Array List class as it provides functionalities
+ * that are unique for duke.
+ */
+public class TaskList extends ArrayList<Task> {
+
+    public TaskList() {
+        super();
+    }
+
+    /**
+     * Function that prints all the tasks currently in the TaskList.
+     *
+     */
+
+    public String printList() {
+        String output = "";
+        for (int i = 0; i < this.size(); i++) {
+            int num = i + 1;
+            output = output + num + ". " + this.get(i) + "\n";
+        }
+        return output;
+    }
+
+    public String findList(String input) {
+        int size = this.size();
+        int index = 1;
+        String output = "";
+        for (int i = 0; i < size; i++) {
+            Task t = this.get(i);
+            if (t.getTask().contains(input)) {
+                output = output + index + ". " + t + "\n";
+                index++;
+            }
+        }
+        return output;
+    }
+
+    /**
+     * Marks a task in the Task List at index as done.
+     *
+     * @param index Index of task user wish to mark.
+     * @return Task that has been marked done.
+     */
```

Comment:
I noticed that some of the function documentation starts right above the function, however some has a one line spacing in between. Perhaps you could consider standardizing them?